llarp/util/str.cpp

```diff
@@ -96,7 +96,7 @@ namespace llarp
         last = next;
 
         // advance to next non-delimiter
-        while (str[last] == delimiter)
+        while (str[last] == delimiter and last < str_size)
           last++;
```

Comment:
what were you running into here? i have a lot of tests around `split()` -- i'd be happy to add a regression test